### ISSUE TITLE
How add new loss function?

### ISSUE THREAD

**Issue by hlee13** [Microsoft/LightGBM#623]
Such as the title. 

Only found the Metric class. Can not found the gradient definition corresponding to metric。
l1, l2, l2_root

So what need I do. Only add a new metric class?

**Comment by guolinke** [comment]
metric is not the loss function.
You can inherit the objective class, which is the loss function.
